### PR TITLE
find: various improvements

### PR DESCRIPTION
**Pull request by waldyrious**
The information content of the page should be unchanged, but hopefully it's now easier to read
(and respects our length recommendations of 8 maximum examples).

The changes include:
- Make the `-mtime` example more intuitive, so it only needs to be shown once,
  and combine the `-delete` option with the first (now only) `-mtime` example
  (thus reducing the page size back to 8 examples)
- Fix a comma splice in the `-exec` example, and a missing "a" in the `-path` example
- Reorder the examples to provide a more consistent and intuitive sequence
  (e.g. by introducing `-iname` in an earlier example, and moving the `-exec` command down)

- [ ] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [ ] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

**Comment by waldyrious** [comment]
Note to whoever merges this: please make sure to edit the extended commit message to add the missing closing backtick around "-delete".

**Comment by agnivade** [comment]
> Note to whoever merges this: please make sure to edit the extended commit message to add the missing closing backtick around "-delete".

Can you not just force-push this to fix it ?

**Comment by waldyrious** [comment]
> Can you not just force-push this to fix it ?

Yes, I can now :) when I opened the PR I was on a computer without git set up so just figured I'd leave the note just in case.

File: pages/common/find.md [hunk 1]
```diff
@@ -10,30 +10,26 @@
...
 
-`find {{root_path}} -name '{{*.ext}}' -exec {{wc -l {} }}\;`
-
-- Find files modified in the last 24-hour period:
-
-`find {{root_path}} -mtime {{-1}}`
+`find {{root_path}} -name '{{*.py}}' -not -path '{{*/site-packages/*}}'`
 
-- Find files using case insensitive name matching, of a certain size:
+- Find files of a certain size:
```
Comment (mebeim) [inline pages/common/find.md]: I would suggest rewording this to better clarify that those are lower and upper limits and not exact sizes.

Comment (waldyrious) [reply pages/common/find.md]: Yeah, now that the size example is not overloaded with the case-insensitive variation, the description can be a bit more explicit. 👍

Will change.

Comment (waldyrious) [reply pages/common/find.md]: Done!

**Comment by sbrl** [comment]
That's a good point, actually. when merging, should the extended commit message stay? Under what circumstances should it stay or be removed?

**Comment by mebeim** [comment]
@sbrl well that's up to us. I think it is helpful to have an extended commit message if the author is willing to write one like in this case.

**Comment by waldyrious** [comment]
> when merging, should the extended commit message stay? Under what circumstances should it stay or be removed?

> I think it is helpful to have an extended commit message if the author is willing to write one like in this case. I remember squashing some PRs in the past by copy-pasting the commit message to keep it.

@sbrl, @mebeim the second-to-last item of the ["Handling PRs" section](https://github.com/tldr-pages/tldr/blob/master/contributing-guides/maintainers-guide.md#ii-handling-prs) of our Maintainers' guide addresses precisely this:

> When merging PRs, use the **merge strategy that produces a clean git history**:
> - If there's a single commit in the PR, or if the multiple commits are not semantically independent changes, use the `Squash and merge` method.
(Don't forget to clean up the body of the squashed commit message.)
> - If instead the PR author took the time to craft individual, informative messages for each commit, then use the `Rebase and merge` method, to honor that work and preserve the history of the changes.
> - For less clear-cut cases, a simple heuristic you can follow is that if there are more "dirty" commits than "clean" commits, then prefer squash, else do a rebase.

(split into bullet points here, for readability)

I actually think that may not sufficiently explicit, so if you can think of a way to make it less ambiguous for you, please do edit it :)

**Comment by waldyrious** [comment]
In short, I think that, as a general rule, extended messages should be kept, unless they don't add any useful information -- for example, those automatically generated by github when a squash merge is being done.

**Comment by waldyrious** [comment]
@andrik for the record, according to the conversation above, I'd have preferred the commit message to have been cleaned up a bit. Instead of:

> **find: various improvements (#3124)**
>
>\* find: various improvements
>
>The information content of the page should be unchanged, but hopefully it's now easier to read
>(and respects our length recommendations of 8 maximum examples).
>
>The changes include:
>\- Make the `-mtime` example more intuitive, so it only needs to be shown once,
>  and combine the `-delete` option with the first (now only) `-mtime` example
>  (thus reducing the page size back to 8 examples)
>\- Fix a comma splice in the `-exec` example, and a missing "a" in the `-path` example
>\- Reorder the examples to provide a more consistent and intuitive sequence
>  (e.g. by introducing `-iname` in an earlier example, and moving the `-exec` command down)
>
>\* clarify -size example description

it should be simply

> **find: various improvements (#3124)**
>
>The information content of the page should be unchanged, but hopefully it's now easier to read
>(and respects our length recommendations of 8 maximum examples).
>
>The changes include:
>\- Make the `-mtime` example more intuitive, so it only needs to be shown once,
>  and combine the `-delete` option with the first (now only) `-mtime` example
>  (thus reducing the page size back to 8 examples)
>\- Fix a comma splice in the `-exec` example, and a missing "a" in the `-path` example
>\- Reorder the examples to provide a more consistent and intuitive sequence
>  (e.g. by introducing `-iname` in an earlier example, and moving the `-exec` command down)

Does that make sense?

**Comment by mebeim** [comment]
Oh yes, that's what should've been done. I guess you can't edit anymore unfortunately? Or maybe @sbrl can with their owner super powers, but I don't know if force-pushing master is a good idea in the first place.

**Comment by waldyrious** [comment]
Yeah, it could be done by force-pushing to master, but the situation doesn't justify such a drastic measure. Our git history is already much messier than that anyway! 😄
